File: leet_083_sorted_lst_del_dup.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head)
    {
        ListNode    *fast, *slow;

        if (!head)
            return head ;
        fast = head->next;
        slow = head ;
        while (fast)
        {
            if (slow->val ^ fast->val)
            {
                slow->next = fast;
                slow = slow->next;
                fast = fast->next;
            }
            else
                fast = fast->next;
        }
        slow->next = fast;
        return head ;
    }
};

File: leet_083_sorted_lst_del_dup.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None
        fast = head.next
        slow = head
        while fast:
            if slow.val == fast.val:
                fast = fast.next
            else:
                slow.next = fast
                slow = slow.next
                fast = fast.next
        slow.next = fast
        return head
